app/core/models/ocelot/stanzas/CalculationStanza.scala

```diff
@@ -102,10 +129,16 @@ sealed trait Operation {
 
   }
 
+  def listAndValueOp(list: List[String], scalar: String, f:(List[String], String) => Option[List[String]], labels: Labels): Labels =
+  f(list, scalar) match {case Some(value) => labels.updateList(label, value) case None => labels}
+
+  def listOp(list1: List[String], list2: List[String], f:(List[String], List[String]) => List[String], labels: Labels): Labels =
+    labels.updateList(label, f(list1, list2))
+
   def rounding(f: (BigDecimal, Int) => BigDecimal, labels: Labels): Labels = {
 
-    val x: String = value(left, labels)
-    val y: String = value(right, labels)
+    val x: String = value(left, labels).getOrElse("")
```

Comment:
Rather than getOrElse("") etc, if either of the values are missing return immediately